### PR TITLE
Another round of coverity fixes

### PR DESCRIPTION
**Pull request by nschloe**

### CODE REVIEW

File: src/controllers/qtscript-bytearray/bytearrayclass.cpp [hunk 1]
```diff
@@ -135,7 +135,6 @@ QScriptValue ByteArrayClass::property(const QScriptValue &object,
             return QScriptValue();
         return uint(ba->at(pos)) & 255;
     }
-    return QScriptValue();
```
Comment (daschuer) [inline src/controllers/qtscript-bytearray/bytearrayclass.cpp]: the else above is also redundant

Comment (nschloe) [inline src/controllers/qtscript-bytearray/bytearrayclass.cpp]: I don't see this from the code. Should I go ahead remove the `else` code?

File: src/widget/wlibrary.cpp [hunk 2]
```diff
@@ -39,7 +39,12 @@ void WLibrary::switchToView(const QString& name) {
         if (widget != NULL && currentWidget() != widget) {
             //qDebug() << "WLibrary::setCurrentWidget" << name;
             setCurrentWidget(widget);
-            dynamic_cast<LibraryView*>(widget)->onShow();
+            LibraryView * lview = dynamic_cast<LibraryView*>(widget);
+              if (lview == NULL) {
```
Comment (daschuer) [inline src/widget/wlibrary.cpp]: The dynamic cast should be done above the if statement above `if (widget !=..`
Please check also indent and brake around column ~80

File: src/controllers/qtscript-bytearray/bytearrayclass.cpp (hunk 1 above)
Comment (daschuer) [inline src/controllers/qtscript-bytearray/bytearrayclass.cpp]: my idea was this: 
``` 
if (name == length) {
    return ba->length();
}
qint32 pos = id;
if ((pos < 0) || (pos >= ba->size())) {
    return QScriptValue();
}
return uint(ba->at(pos)) & 255;

```

Comment (nschloe) [inline src/controllers/qtscript-bytearray/bytearrayclass.cpp]: I see. Well, it doesn't matter much which we do. I'll check in your suggestion.

File: src/widget/wlibrary.cpp (hunk 2 above)
Comment (nschloe) [inline src/widget/wlibrary.cpp]: > above the if statement 

Why? If `widget == NULL` (or the other is false), we don't need to cast at all.

File: src/controllers/controllerengine.cpp [hunk 3]
```diff
@@ -1198,7 +1198,13 @@ double ControllerEngine::getDeckRate(const QString& group) {
 
 bool ControllerEngine::isDeckPlaying(const QString& group) {
     ControlObjectThread* pPlay = getControlObjectThread(group, "play");
-    return pPlay->get() > 0.0;
+    if (pPlay) {
```
Comment (ywwg) [inline src/controllers/controllerengine.cpp]: It's better to check for error condition first, like this:

if (pPlay == NULL) {
   handle error
  return
}

return pPlay->...

That style usually results in much nicer code in general.

File: src/encoder/encodervorbis.cpp [hunk 4]
```diff
@@ -42,6 +42,34 @@ EncoderVorbis::EncoderVorbis(EncoderCallback* pCallback)
           m_metaDataTitle(NULL),
           m_metaDataArtist(NULL),
           m_metaDataAlbum(NULL){
+  m_vdsp.pcm_returned = 0;
```
Comment (ywwg) [inline src/encoder/encodervorbis.cpp]: please check indentation

File: src/library/dao/analysisdao.cpp [hunk 5]
```diff
@@ -347,9 +347,6 @@ void AnalysisDao::saveTrackAnalyses(TrackInfoObject* pTrack) {
                  << "waveform analysis for trackId" << trackId
                  << "analysisId" << analysis.analysisId;
 
-    if (pWaveSummary->getId() != -1) {
```
Comment (ywwg) [inline src/library/dao/analysisdao.cpp]: what happened to this?

Comment (nschloe) [inline src/library/dao/analysisdao.cpp]: The value of `analysis.analysisId` gets overridden right after.

Comment (ywwg) [inline src/library/dao/analysisdao.cpp]: maybe that's a bug?

File: src/widget/wlibrary.cpp (hunk 2 above)
Comment (daschuer) [inline src/widget/wlibrary.cpp]: Right. We want also not setCurrentWidget() if it's not a LibraryView

Comment (daschuer) [inline src/widget/wlibrary.cpp]: if you like you could also fix the double lookup using contains() and [] some lines above
Just use a single QMap::value(const Key & key, const T & defaultValue = T()) call.

File: src/library/dao/analysisdao.cpp (hunk 5 above)
Comment (nschloe) [inline src/library/dao/analysisdao.cpp]: The comment indicates this is intentional.

Comment (ywwg) [inline src/library/dao/analysisdao.cpp]: ah ok now I see the context.

**Comment by daschuer** [comment]
Thank you for the work! LGTM 
Please give a sign if you are ready.

**Comment by nschloe** [comment]
Good to go.

**Comment by nschloe** [comment]
We might better wait for travis to give a green light here, but if we cross our fingers it'll run through fine this time. :)

**Pull request merged**